### PR TITLE
fix(wallet): Move Portfolio Settings to Wallet Settings Menu

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
- Moves the `Portfolio` settings to the `Wallet` settings menu
- Replaces the `Portfolio` settings button with a `Plus` button to open `Edit Visible Assets` modal

Resolves <https://github.com/brave/brave-browser/issues/37257>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open the `Wallet` and navigate to the `Portfolio` screen
2. Open the `Wallet Settings` menu, you should now see the `Portfolio Settings` in there.
3. Click on the the `Plus` button in the `Portfolio` header, it should open the `Edit Visible Assets` modal

https://github.com/brave/brave-core/assets/40611140/d7bb083f-963b-4539-9b6f-f5a02c8abf21

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/card-headers/portfolio-overview-header.tsx [hunk 1]
```diff
@@ -15,55 +19,38 @@ import { DefaultPanelHeader } from './default-panel-header'
...
-      padding='24px 0px'
-      justifyContent='space-between'
-    >
-      <HeaderTitle>{getLocale('braveWalletTopNavPortfolio')}</HeaderTitle>
-      <MenuWrapper ref={portfolioOverviewMenuRef}>
-        <MenuButton
-          onClick={() => setShowPortfolioOverviewMenu((prev) => !prev)}
-        >
-          <ButtonIcon name='tune' />
+    <>
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/card-headers/portfolio-overview-header.tsx]: Good catch! Removed!

File: components/brave_wallet_ui/components/desktop/wallet-menus/wallet_settings_menu.tsx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+  const onToggleHideNFTsTab = React.useCallback(() => {
+    if (walletLocation.includes(WalletRoutes.PortfolioNFTs)) {
+      history.push(WalletRoutes.PortfolioAssets)
+    }
+    setHidePortfolioNFTsTab((prev) => !prev)
+  }, [history, setHidePortfolioNFTsTab, walletLocation])
+
+  const onClickRoute = (route: WalletRoutes | AccountPageTabs) => {
+    if (route === WalletRoutes.AddHardwareAccountModalStart) {
+      chrome.tabs.create({ url: `chrome://wallet${route}` }, () => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/wallet-menus/wallet_settings_menu.tsx:144]: Yep good catch! Fixed!

File: components/brave_wallet_ui/components/desktop/wallet-menus/wellet-menus.style.ts [hunk 3]
```diff
@@ -15,12 +15,13 @@ import {
 export const StyledWrapper = styled.div<{
   yPosition?: number
   right?: number
+  noPadding?: boolean
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/wallet-menus/wellet-menus.style.ts]: Updated!

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Review by josheleonard — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**